src/http_server.cc

```diff
@@ -3233,8 +3237,11 @@ HTTPAPIServer::HandleGenerate(
 
   // If tracing is enabled see if this request should be traced.
   TRITONSERVER_InferenceTrace* triton_trace = nullptr;
-  std::shared_ptr<TraceManager::Trace> trace =
-      StartTrace(req, model_name, &triton_trace);
+  std::shared_ptr<TraceManager::Trace> trace;
+  if (trace_manager_) {
```

Comment:
Why not put into `StartTrace()`

Reply:
Will make these changes as well as a part of this refactoring ticket [DLIS-7380].